src/game/shell.c

```diff
@@ -182,6 +188,78 @@ void Shell_Wait(int nticks)
     }
 }
 
+void Shell_GetScreenshotName(char *str)
+{
+    assert(str);
+
+    // Get level title of unknown length
+    char level_title[100];
+    snprintf(
+        level_title, LEVEL_TITLE_SIZE, "%s",
+        g_GameFlow.levels[g_CurrentLevel].level_title);
+
+    // Trim level titles that are too long
+    if (strlen(level_title) >= LEVEL_TITLE_SIZE) {
+        level_title[LEVEL_TITLE_SIZE] = '\0';
+    }
+
+    // Prepare level title for screenshot
+    char *check = level_title;
+    bool prev_us = true; // '_' after timestamp before title
+    int idx = 0;
+
+    while (*check != '\0') {
+        if (*check == ' ') {
+            // Replace spaces with a single underscore
+            if (prev_us) {
+                memmove(
+                    &level_title[idx], &level_title[idx + 1],
+                    strlen(level_title) - idx);
+            } else {
+                *check++ = '_';
+                idx++;
+                prev_us = true;
+            }
+        } else if (
+            ((*check < 'A' || *check > 'Z') && (*check < 'a' || *check > 'z')
+             && (*check < '0' || *check > '9'))
+            || *check == '\'' || *check == '.') {
```

Comment:
I think this should be `&& *check != '\'' && check != '.'`?

Reply:
I posted a comment above, but I don't think I need this at all anymore. We don't want `'` or `.` in the screenshot name right? I think I just need:

```
        } else if (
            ((*check < 'A' || *check > 'Z') && (*check < 'a' || *check > 'z')
             && (*check < '0' || *check > '9'))
```
?

Reply:
I mean if we want to strip these characters then we don't need these checks altogether. If we want to keep them, they need to be reversed.

Reply:
Well keeping `.` in the screenshot name could potentially screw up the file extension?

Also we still want to strip any spaces, and symbols like `< > ? !` etc right?